src/libraries/Native/Unix/CMakeLists.txt

```diff
@@ -100,7 +100,8 @@ if(CLR_CMAKE_TARGET_ANDROID)
 endif()
 
 string(TOUPPER ${CMAKE_BUILD_TYPE} UPPERCASE_CMAKE_BUILD_TYPE)
-if (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG)
+
+if (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG OR UPPERCASE_CMAKE_BUILD_TYPE STREQUAL CHECKED)
     add_compile_options(-O0)
```

Comment:
Should optimizations stay enabled for CHECKED builds?

Reply:
Yes. I suggested earlier to just map Checked --> Debug for simplicity. But since we are configuring optimizations right here, where we still know it is CHECKED, it would be preferable to have optimizations on.

Reply:
Yes, that would make sense. Frankly, the libraries cmakelists.txt duplicate all the compiler settings that we use in coreclr and installer and have extracted them some time ago into `eng/configurecompiler.cmake`, so it would be beneficial to clean things up and use that in libraries too.